src/components/panel/panel.js

```diff
@@ -945,6 +955,31 @@ MdPanelRef.prototype.detach = function() {
   return this._detachPromise;
 };
 
+/**
+ * Destroys the panel.
+ *
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * destroyed.
+ */
+MdPanelRef.prototype.destroy = function() {
+  if (this.isDestroyed) {
+    return this._destroyPromise;
+  }
+
+  var self = this;
+
+  var destroyFn = function () {
+    self.isDestroyed = true;
+    self._config.locals = null;
+  };
+
+  this._destroyPromise = this._$q.when(destroyFn())
```

Comment:
Why does destroy need to be a promise? This file is already promise heavy and everything you are doing within destroy is synchronous.

If it needs to be a promise, then you should stick to one of the promise styles above instead of introducing a 3rd variation. So,
```
var self = this;
this._destoryPromise = this._$q(function(resolve) {
  self.isDestroyed = true;
  self._config.locals = null;
  resolve(self);
});
```
Nothing can cause this to reject, so you can ignore that part.